### ISSUE TITLE
Immediate exit when dragging something over the window.

### ISSUE THREAD

**Issue by slembcke** [buserror/mii_emu#11]
If I drag a file or browser tab over the window, it immediately exits. According to GDB it exits normally.

**Comment by buserror** [comment]
Hmm here I use cinnamon so full screen GL really; I can drag windows over/under all I want and it's fine.... are you using 'nouveau' or some other driver?

**Comment by slembcke** [comment]
Fedora/XWayland/AMD here. To be clear, I don't mean dragging windows over it, but _drag and drop_ objects. So by "browser tab" I mean when you pop out a Firefox tab, that's a droppable object. Same with dragging a file icon over the window, it just immediately quits.

**Comment by slembcke** [comment]
Following up, it's this "client message" line that causing the exit:
https://github.com/buserror/mii_emu/blob/2e12a37bceebef3787abddc1ca10d82fdc427e9d/ui_gl/mii_emu_gl.c#L678

I've never done X programming directly before, but here's what gdb has to say about `evt.xclient` if that's helpful anyway:
```
(gdb) p evt.xclient
$2 = {type = 33, serial = 371, send_event = 1, display = 0xa5e0b0, window = 23068674, message_type = 307, format = 32, data = {
    b = "\004\000`\000\000\000\000\000\001\000\000\005\000\000\000\000\000\000\000", s = {4, 96, 0, 0, 1, 1280, 0, 0, 0, 0}, l = {6291460, 83886081, 0, 0, 0}}}
```

**Comment by slembcke** [comment]
Throwing this in there:
`printf("atom_name: %s\n", XGetAtomName(ui->dpy, evt.xclient.message_type));`

Results in the following which isn't too surprising:
`atom_name: XdndEnter`

Is there a particular event you are looking for there?

**Comment by buserror** [comment]
Interesting; it must be remnant of some other code. The Client Message can be used for quite a few things, wonder why it's still there! Good catch tho, I obviously never really look at that code :-)
Also, strangely, it never happens on Xorg! -- I'll remove these lines...

**Comment by buserror** [comment]
Ahh I remember now, it CAN be used to prevent to have better granularity when scaling the window, but I had found another way to do it anyway so completely unneeded. Good find :-)

**Comment by buserror** [comment]
Oh, I ALSO figured out why it was still there, and in fact fixed it. ClientMessage are used for the closebox of the window manager, so it was there to allow closing the window, but of course that did it for any ClientMessage. I've now added proper filtering. Nice catch :-)

**Comment by slembcke** [comment]
Seems to be fixed in 1.8.
